pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -46,16 +47,32 @@ const (
 
 const loadBalancerProvisioningStatusActive = "ACTIVE"
 
+// We wrap the net.LookupHost function in a variable to allow overriding it in unit tests.
+//
+//nolint:gocritic
+var lookupHost = func(host string) ([]string, error) {
+	return net.LookupHost(host)
+}
+
 func (s *Service) ReconcileLoadBalancer(openStackCluster *infrav1.OpenStackCluster, clusterName string, apiServerPort int) (bool, error) {
 	loadBalancerName := getLoadBalancerName(clusterName)
 	s.scope.Logger().Info("Reconciling load balancer", "name", loadBalancerName)
 
 	var fixedIPAddress string
+	var err error
+
 	switch {
 	case openStackCluster.Spec.APIServerFixedIP != "":
 		fixedIPAddress = openStackCluster.Spec.APIServerFixedIP
 	case openStackCluster.Spec.DisableAPIServerFloatingIP && openStackCluster.Spec.ControlPlaneEndpoint.IsValid():
-		fixedIPAddress = openStackCluster.Spec.ControlPlaneEndpoint.Host
+		ips, err := lookupHost(openStackCluster.Spec.ControlPlaneEndpoint.Host)
+		if err != nil {
+			return false, fmt.Errorf("lookup host: %w", err)
+		}
+		fixedIPAddress = ips[0]
+		if net.ParseIP(fixedIPAddress) == nil {
+			s.scope.Logger().Info("ControlPlaneEndpoint.Host is not an IP address, using the first resolved IP address", "host", openStackCluster.Spec.ControlPlaneEndpoint.Host, "ip", fixedIPAddress)
+		}
```

Comment:
This code block feels slightly too long to be duplicated below. Could we DRY this somehow?

Reply:
I've put it into a constant. Let me know if that works.